File: RobWork/example/java/src/ExRotation3D.java
import org.robwork.LoaderRW;
import org.robwork.sdurw.*;
import static org.robwork.sdurw.sdurw.*;

public class ExRotation3D {
    public static void main(String[] args) throws Exception {
        LoaderRW.load("sdurw");

        Rotation3Dd rotd = new Rotation3Dd(1,0,0,0,0,-1,0,1,0);
        Rotation3Df rotf = new Rotation3Df(1,0,0,0,0,-1,0,1,0);

        System.out.println("Rotation double:");
        System.out.println(rotd);
        System.out.println("Rotation float:");
        System.out.println(rotf);
        System.out.println("Rotation inverse:");
        System.out.println(rotd.inverse());
        System.out.println("Identity:");
        System.out.println(rotd.multiply(inverse(rotd)));
    }
}

File: RobWork/example/lua/ex-rotation3d.lua
require("sdurw_core")
require("sdurw")
using("sdurw")
require("sdurw_math")
using("sdurw_math")

rotd = Rotation3Dd(1,0,0,0,0,-1,0,1,0);
rotf = Rotation3Df(1,0,0,0,0,-1,0,1,0);

print("Rotation double:");
print(tostring(rotd));
print("Rotation float:");
print(tostring(rotf));
print("Rotation inverse:");
print(tostring(inverse(rotd)));
print("Identity:");
print(tostring(rotd*inverse(rotd)));